shared/common/ui/switch/index.tsx

```diff
@@ -11,23 +11,29 @@ export interface SwitchProps {
   rightLabel: string;
   onClick: () => void;
   classes?: string;
-  defaultState?: SwitchState;
+  defaultState?: switchState;
+  disabled?: boolean;
 }
 
 export const Switch = ({
-  defaultState = SwitchState.left,
+  defaultState = switchState.left,
   leftLabel,
   rightLabel,
   onClick,
-  classes,
+  disabled = false
 }: SwitchProps) => {
   const [leftActive, setLeftActive] = useState(
-    defaultState === SwitchState.left
+    defaultState === switchState.left
+  );
+
+  useEffect(
+    () => setLeftActive(defaultState === switchState.left),
+    [defaultState]
```

Comment:
I don't think the `Switch` component should have this internal state that it then has to sync with the `defaultState` value. It should be just like a controlled component. So I would do something like this:

```tsx
export interface SwitchProps<T> {
  labels: [string, string];
  values: [T, T];
  selected: T;
  onChange: (value: T) => void;
  disabled?: boolean;
  className?: string;
}

export function Switch<T>({
  labels,
  values,
  selected,
  onChange,
  disabled = false,
  className,
}: SwitchProps<T>) {
  const leftActive = selected === values[0];

// ...
  <label>
    {labels[0]}
    <input checked={leftActive} onChange={() => onChange(values[0])} />
  </label>
// ...
  <label>
    {labels[1]}
    <input checked={!leftActive} onChange={() => onChange(values[1])} />
  </label>
```

Reply:
omg ofc, I don't know why I didn't make it like a normal controlled component, I focused too much on design
tnx